samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemAdmin.scala

```diff
@@ -38,6 +39,8 @@ import scala.collection.JavaConverters._
 
 object KafkaSystemAdmin extends Logging {
 
+  // Used only for test
+  @volatile var deleteMessagesCalled = false
   val CLEAR_STREAM_RETRIES = 3
 
   /**
```

Comment:
Just one question: I thought that the introduction of AdminClient would replace our need to instantiate any SimpleConsumer instance, or using KafkaUtils to get topic metadata and fetch offsets? It could be addressed in a separate PR, but I just want to make sure that's the plan and there is no blockers to do that.

Reply:
Thanks for all the review @nickpan47! Yes, AdminClient can be used to get topic metadata. But we would need to use KafkaConsumer.offsetsForTimes to fetch offset for e.g. earliest offset or latest offset.